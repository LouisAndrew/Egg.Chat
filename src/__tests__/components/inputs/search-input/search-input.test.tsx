import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import ChatInput from 'components/inputs/search-input';

describe('ChatInput', () => {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const mockSearch = jest.fn((name) => {});

    const Element = <ChatInput search={mockSearch} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
        expect(true).toBeTruthy();
    });

    it('Should render the necessary fields', () => {
        const { queryByPlaceholderText } = render(Element);
        const inputField = queryByPlaceholderText('SEARCH NAME');

        expect(inputField).toBeInTheDocument();
    });

    it('Should search for the available / in-contact chatroom when user types in the input field', () => {
        const { getByPlaceholderText } = render(Element);

        const inputField = getByPlaceholderText('SEARCH NAME');

        const mockInput = 'John Doe';

        userEvent.type(inputField, mockInput);
        expect(mockSearch).toBeCalled();
        expect(mockSearch).toBeCalledWith(mockInput);
    });

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            // eslint-disable-next-line jest/no-conditional-expect
            expect(tree).toMatchSnapshot();
        }
    });
});
