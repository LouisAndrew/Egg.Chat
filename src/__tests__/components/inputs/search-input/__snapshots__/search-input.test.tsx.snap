// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput matches snapshot 1`] = `
<form
  className="css-vurnku"
  onSubmit={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
      fillRule="evenodd"
    />
  </svg>
  <input
    className="css-4y926w"
    onChange={[Function]}
    placeholder="SEARCH NAME"
    type="text"
  />
</form>
`;
